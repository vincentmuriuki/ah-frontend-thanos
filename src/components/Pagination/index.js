import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

const Pagination = ({ article, pageNumber }) => {
  const activePage = number => (number === Number(pageNumber) ? 'active' : null);
  const disablePrevious = (num1, num2) => (num1 === num2 ? 'disabled' : null);

  const numberOfpages = Math.ceil(article.count / 10);
  const ArraynumberOfpages = _.range(1, numberOfpages + 1);
  const pages = ArraynumberOfpages.map(number => (
    <li key={number} className={`page-item ${activePage(number)}`}>
      {' '}
      <a className="page-link" href={`articles/page/${number}`}>{number}</a>
    </li>
  ));

  return (
    <div>
      <br />
      <ul className="pagination justify-content-center">
        <li className={`page-item ${disablePrevious(pageNumber, '1')}`}>
          <a className="page-link" href={`articles/page/${Number(pageNumber) - 1}`} tabIndex="-1">Previous</a>
        </li>
        {pages}
        <li className={`page-item ${disablePrevious(Number(pageNumber), numberOfpages)}`}>
          <a className="page-link" href={`articles/page/${Number(pageNumber) + 1}`}>Next</a>
        </li>
      </ul>
    </div>
  );
};

Pagination.propTypes = {
  article: PropTypes.shape({}).isRequired,
  pageNumber: PropTypes.string.isRequired,
};

export default Pagination;
