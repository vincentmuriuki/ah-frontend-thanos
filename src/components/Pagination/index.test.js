import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '.';


describe('Test  pagination', () => {
  const article = {
    count: 4,
  };
  const pageNumber = '3';
  const Paginationwrapper = shallow(
    <Pagination article={article} pageNumber={pageNumber} />,
  );

  test('Snap shot pagination Component', () => {
    expect(Paginationwrapper).toMatchSnapshot();
  });
  test('Pagination should render correctly', () => {
    expect(Paginationwrapper.find('.pagination').length).toBe(1);
  });
});
