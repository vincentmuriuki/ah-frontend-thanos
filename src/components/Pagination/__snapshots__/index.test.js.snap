// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test  pagination Snap shot pagination Component 1`] = `
<div>
  <br />
  <ul
    className="pagination justify-content-center"
  >
    <li
      className="page-item null"
    >
      <a
        className="page-link"
        href="articles/page/2"
        tabIndex="-1"
      >
        Previous
      </a>
    </li>
    <li
      className="page-item null"
      key="1"
    >
       
      <a
        className="page-link"
        href="articles/page/1"
      >
        1
      </a>
    </li>
    <li
      className="page-item null"
    >
      <a
        className="page-link"
        href="articles/page/4"
      >
        Next
      </a>
    </li>
  </ul>
</div>
`;
